packages/grid/x-data-grid-premium/src/models/dataGridPremiumProps.ts

```diff
@@ -153,4 +153,9 @@ export interface DataGridPremiumPropsWithoutDefaultValue<R extends GridValidRowM
     cellSelectionModel: GridCellSelectionModel,
     details: GridCallbackDetails,
   ) => void;
+  /**
+   * Callback fired when the state of the Excel export changes.
+   * @param {boolean} inProgress Indicates if the task is in progress.
+   */
+  onExcelExportStateChange?: (inProgress: boolean) => void;
```

Comment:
What about using string for this API. It's hard to remember if the boolean means "in progress" or "task done"

```suggestion
  onExcelExportStateChange?: (status: 'pending' | 'finished') => void;
```